src/core/global.c

```diff
@@ -190,6 +179,10 @@ static void nn_global_handler (struct nn_fsm *self,
 static void nn_global_shutdown (struct nn_fsm *self,
     int src, int type, void *srcptr);
 
+/*  Socket holds. */
+static int nn_global_hold_socket(struct nn_sock **sockp, int s);
+static int nn_global_hold_socket_locked(struct nn_sock **sockp, int s);
+static void nn_global_rele_socket(struct nn_sock *);
```

Comment:
> Acquire/release feels a little more conventional

Wait a minute.... acquire/release is a different semantic than hold/release ... yes? They are not synonyms.

Acquire/release is for mutexing; hold/release is for reference counting.

Don't mind my comments on terminologies so much on this review; I need to just grok a bit better. At this point, it feels the code is correct and I'm needing to catch up.